### PR TITLE
Automatically wrap Pointers in Arrays

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: examples/snap/snap.jl [hunk 1]
```diff
@@ -40,16 +40,12 @@ A = LMP(["-screen","none"]) do lmp
         command(lmp, "dump 2 all custom 100 dump.forces fx fy fz")
         command(lmp, "run 0")
 
-        # LAMMPS returns this as double...
-        natoms = Int(get_natoms(lmp))
+        nlocal = extract_global(lmp, "nlocal")
 
         types = extract_atom(lmp, "type", LAMMPS.API.LAMMPS_INT)
```
Comment (vchuravy) [inline examples/snap/snap.jl:45]: @rohskopf one thing I noticed is that the types are not ordered -- not all type 1 atoms are listed as the first 96 atoms.

File: examples/snap/snap.jl [hunk 2]
```diff
@@ -62,29 +58,33 @@ A = LMP(["-screen","none"]) do lmp
...
+        for k in 1:(ncoeff-1)
             acc = 0.0
             for n in 1:N1
-                acc += bs[n][k]
+                # n = permutation[n]
+                acc += bs[n, k]
             end
             push!(row, acc)
         end
+        @show row
```
Comment (vchuravy) [inline examples/snap/snap.jl:80]:
```suggestion
```

File: src/LAMMPS.jl [hunk 3]
```diff
@@ -76,14 +71,100 @@ function extract_atom(lmp::LMP, name, dtype::Union{Nothing, API._LMP_DATATYPE_CO
...
+    if length(shape) > 1
+        # We got a list of ptrs,
+        # but the first pointer points to the whole data
+        ptr = Base.unsafe_load(ptr)
+
+        @assert length(shape) == 2
+
+        # Note: Julia like Fortran is column-major
+        #       so the data is transposed from Julia's perspective
+        shape = reverse(shape)
```
Comment (vchuravy) [inline src/LAMMPS.jl:111]: Also see https://github.com/lammps/lammps/blob/1e5e08fc1bd9ebd4428e0a420ef0055b132afbef/examples/COUPLE/fortran2/LAMMPS.F90#L900

File: examples/snap/snap.jl [hunk 4]
```diff
@@ -19,7 +19,7 @@ const ncoeff = round(Int, (J+1)*(J+2)*((J+(1.5))/3.) + 1)
 A = LMP(["-screen","none"]) do lmp
 
     M = 61
-    A = Array{Float64}(undef, M, 2*(ncoeff+1))
+    A = Array{Float64}(undef, M, 2*ncoeff) # bispectrum is 2*(ncoeff - 1) + 2
```
[the remark this answers is not in the document]
Comment (rohskopf) [reply examples/snap/snap.jl:22]: @vchuravy yes that's correct. I'll think about how to make this code more general, it's all very hard-coded as is.

**Pull request merged**